protocol/x/clob/e2e/batch_cancel_test.go

```diff
@@ -281,3 +289,527 @@ func TestBatchCancelSingleCancelFunctionality(t *testing.T) {
+						{
+							ClobPairId: 0,
+							ClientIds:  []uint32{0, 3},
+						},
+						{
+							ClobPairId: 1,
+							ClientIds:  []uint32{1, 2, 4, 5},
+						},
+					},
+					GoodTilBlock: 30,
+				},
+			},
+			expectedOrderIdsInMemclob: map[clobtypes.OrderId]bool{
+				PlaceOrder_Alice_Num1_Id0_Clob0_Sell10_Price10_GTB20.Order.OrderId:     false,
+				constants.Order_Alice_Num1_Id1_Clob1_Sell10_Price15_GTB20.OrderId:      false,
+				PlaceOrder_Alice_Num1_Id2_Clob1_Buy10_Price10_GTB26.Order.OrderId:      false,
+				constants.Order_Alice_Num1_Id3_Clob0_Sell100_Price100000_GTB20.OrderId: false,
+				constants.Order_Alice_Num1_Id4_Clob1_Sell10_Price15_GTB20_PO.OrderId:   false,
+				PlaceOrder_Alice_Num1_Id5_Clob1_Buy10_Price15_GTB23.Order.OrderId:      false,
+			},
+			expectedCancelExpirationsInMemclob: map[clobtypes.OrderId]uint32{
+				constants.Order_Alice_Num1_Id0_Clob0_Sell10_Price10_GTB20.OrderId:      30,
+				constants.Order_Alice_Num1_Id1_Clob1_Sell10_Price15_GTB20.OrderId:      30,
+				PlaceOrder_Alice_Num1_Id2_Clob1_Buy10_Price10_GTB26.Order.OrderId:      30,
+				constants.Order_Alice_Num1_Id3_Clob0_Sell100_Price100000_GTB20.OrderId: 30,
+				constants.Order_Alice_Num1_Id4_Clob1_Sell10_Price15_GTB20_PO.OrderId:   30,
+				PlaceOrder_Alice_Num1_Id5_Clob1_Buy10_Price15_GTB23.Order.OrderId:      30,
+			},
+		},
+		"Batch cancels prevent new orders with lower gtb from being placed": {
+			firstBlockOrders: []clobtypes.MsgPlaceOrder{
+				PlaceOrder_Alice_Num1_Id0_Clob0_Sell10_Price10_GTB20,
+				PlaceOrder_Alice_Num1_Id1_Clob1_Sell10_Price15_GTB20,
+				PlaceOrder_Alice_Num1_Id3_Clob0_Sell100_Price100000_GTB20,
+			},
+			firstBlockBatchCancel: []clobtypes.MsgBatchCancel{
+				{
+					SubaccountId: PlaceOrder_Alice_Num1_Id0_Clob0_Sell10_Price10_GTB20.Order.OrderId.SubaccountId,
+					ShortTermCancels: []clobtypes.OrderBatch{
+						{
+							ClobPairId: 0,
+							ClientIds:  []uint32{0, 3},
+						},
+						{
+							ClobPairId: 1,
+							ClientIds:  []uint32{1},
+						},
+					},
+					GoodTilBlock: 25,
+				},
+			},
+			secondBlockOrders: map[clobtypes.MsgPlaceOrder]bool{
+				PlaceOrder_Alice_Num1_Id0_Clob0_Sell10_Price10_GTB20:      false,
+				PlaceOrder_Alice_Num1_Id1_Clob1_Sell10_Price15_GTB20:      false,
+				PlaceOrder_Alice_Num1_Id3_Clob0_Sell100_Price100000_GTB20: false,
+			},
+			expectedOrderIdsInMemclob: map[clobtypes.OrderId]bool{
+				PlaceOrder_Alice_Num1_Id0_Clob0_Sell10_Price10_GTB20.Order.OrderId:     false,
+				constants.Order_Alice_Num1_Id1_Clob1_Sell10_Price15_GTB20.OrderId:      false,
+				constants.Order_Alice_Num1_Id3_Clob0_Sell100_Price100000_GTB20.OrderId: false,
+			},
+			expectedCancelExpirationsInMemclob: map[clobtypes.OrderId]uint32{
+				constants.Order_Alice_Num1_Id0_Clob0_Sell10_Price10_GTB20.OrderId:      25,
+				constants.Order_Alice_Num1_Id1_Clob1_Sell10_Price15_GTB20.OrderId:      25,
+				constants.Order_Alice_Num1_Id3_Clob0_Sell100_Price100000_GTB20.OrderId: 25,
+			},
+		},
```

Comment:
can we add a test for 
- cancelling non existent orders
- cancel then place order with higher gtb should succeed